### ISSUE TITLE
SymbolSolver fails with StackOverflow in FieldAccessExpression

### ISSUE THREAD

**Issue by AwesomeLemon** [javaparser/javaparser#1454]
Imagine having local variable with the same name as a field of the class of this variable. So, for example, I encountered the problem in code for linked list. Simplified version is as follows: There's a class Node:
```
class Node {
   Node nextSibling;
   Node prevSibling;
}
```
Then in some method there's local variable ``nextSibling`` of type ``Node``.
So the following code should work fine:
```
nextSibling.prevSibling.nextSibling = null;
```
However, if type of the first ``nextSibling`` is to be solved with ``JavaParserFacade.get(typeSolver).getType(expression)`` (where expression is ``NameExpression`` of the said ``nextSibling``), the endless loop is encountered. 

After much pain and debugging I managed to find the problem here: https://github.com/javaparser/javaparser/blob/master/javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/FieldAccessContext.java#L78

At that line the context is the whole ``FieldAccessExpression`` (that is ``nextSibling.prevSibling.nextSibling``). The ``wrappedNode.getName()`` returns ``nextSibling``, meaning the last ``nextSibling``, not the first. The bug arises exactly now, because we wanted to solve first encounter, and now will be solving the last one. And to solve that, everything to the left must be solved. Which leads us to the first ``nextSibling``. Which leads to the second one. And so on and so forth for eternity.

So, to sum up: when solving the first ``a`` in ``a.b.a``, second ``a`` is solved instead, because they have same names, and there's no way to differentiate, and the solving starts with the rightmost encounter.

**Comment by ftomassetti** [comment]
Thanks for your help debugging this one.

I am not sure how to get this solved.
